python/ray/_private/function_manager.py

```diff
@@ -127,6 +131,18 @@ def load_function_or_class_from_local(self, module_name,
         except Exception:
             return None
 
+    def export_key(self, key):
+        """Export a key so it can be imported by other workers"""
+        pos = self._worker.import_thread.num_imported
```

Comment:
I'm not totally sure about the concurrent behavior here. Is it possible that a recent export has not been received by the importer, so the same `pos` is exported again?

Reply:
it won't since we are using internal kv as a transaction db. It'll check until find one that's not used and will hold the spot.

Reply:
IIUC `export_key()` is called by `FunctionActorManager.export()` which is called by Ray `remote()`. Importing is done in `ray_import_thread`. So `import_thread.num_imported` is read and updated in different threads? Can the following happen?

0. `import_thread.num_imported` is 1.
1. Calling `f.remote()` exports `f`, using `pos=2`.
2. Importing thread did not run and update `import_thread.num_imported`.
3. Calling `g.remote()` exports `g`, again using `pos=2`?

Reply:
Hi @mwtian  here we are using check and set. So 3) will happen, but then pos will become 3.